psi4/src/psi4/dlpno/triples.cc

```diff
@@ -0,0 +1,1317 @@
+    outfile->Printf("     T_CUT_TNO set to %6.3e \n", t_cut_tno_pre);
+    outfile->Printf("     T_CUT_DO  set to %6.3e \n", options_.get_double("T_CUT_DO_TRIPLES_PRE"));
+    outfile->Printf("     T_CUT_MKN set to %6.3e \n\n", options_.get_double("T_CUT_MKN_TRIPLES_PRE"));
+
+    triples_sparsity(true);
+    tno_transform(t_cut_tno_pre);
+    double E_T0_pre = compute_lccsd_t0();
+
+    // Step 2: Compute DLPNO-CCSD(T0) energy with surviving triplets
+    outfile->Printf("\n   Continuing computation with surviving triplets...\n");
+    outfile->Printf("     Eliminated all triples with energy less than %6.3e Eh... \n\n", options_.get_double("T_CUT_TRIPLES_WEAK"));
+    triples_sparsity(false);
+    outfile->Printf("    * Energy Contribution From Screened Triplets: %.12f \n\n", de_lccsd_t_screened_);
+
+    outfile->Printf("     T_CUT_TNO (re)set to %6.3e \n", t_cut_tno);
+    outfile->Printf("     T_CUT_DO  (re)set to %6.3e \n", options_.get_double("T_CUT_DO_TRIPLES"));
+    outfile->Printf("     T_CUT_MKN (re)set to %6.3e \n\n", options_.get_double("T_CUT_MKN_TRIPLES"));
+    
+    tno_transform(t_cut_tno);
+    double E_T0 = compute_lccsd_t0();
+    e_lccsd_t_ = e_lccsd_ + E_T0 + de_lccsd_t_screened_;
+
+    outfile->Printf("    DLPNO-CCSD(T0) Correlation Energy: %16.12f \n", e_lccsd_ + E_T0 + de_lccsd_t_screened_);
+    outfile->Printf("    * DLPNO-CCSD Contribution:         %16.12f \n", e_lccsd_);
+    outfile->Printf("    * DLPNO-(T0) Contribution:         %16.12f \n", E_T0);
+    outfile->Printf("    * Screened Triplets Contribution:  %16.12f \n\n", de_lccsd_t_screened_);
+
+
+    // Step 3: Compute full DLPNO-CCSD(T) energy if NOT using T0 approximation
+
+    if (!options_.get_bool("T0_APPROXIMATION")) {
+        outfile->Printf("\n\n  ==> Computing Full Iterative (T) <==\n\n");
+
+        sort_triplets(E_T0);
+
+        double t_cut_tno_strong_scale = options_.get_double("T_CUT_TNO_STRONG_SCALE");
+        double t_cut_tno_weak_scale = options_.get_double("T_CUT_TNO_WEAK_SCALE");
+        outfile->Printf("     T_CUT_TNO (re)set to %6.3e for strong triples \n", t_cut_tno * t_cut_tno_strong_scale);
+        outfile->Printf("     T_CUT_TNO (re)set to %6.3e for weak triples   \n\n", t_cut_tno * t_cut_tno_weak_scale);
+
+        tno_transform(t_cut_tno);
+        estimate_memory();
+
+        double E_T0_crude = compute_lccsd_t0(true);
+        E_T_ = lccsd_t_iterations();
+        double dE_T = E_T_ - E_T0_crude;
+
+        outfile->Printf("\n");
+        outfile->Printf("    DLPNO-CCSD(T0) energy at looser tolerance: %16.12f\n", E_T0_crude);
+        outfile->Printf("    DLPNO-CCSD(T)  energy at looser tolerance: %16.12f\n", E_T_);
+        outfile->Printf("    * Net Iterative (T) contribution:          %16.12f\n\n", dE_T);
+
+        e_lccsd_t_ += dE_T;
+    }
+
+    double e_scf = reference_wavefunction_->energy();
+    double e_ccsd_t_corr = e_lccsd_t_ + de_weak_ + de_lmp2_eliminated_ + de_dipole_ + de_pno_total_;
```

Comment:
should there be more qcvars so all pieces accessible?

Reply:
Yep! Added more qcvars for all the corrections and contributions in CCSD and (T)!